### PR TITLE
Split and deprecate AbstractQuery#useResultCache()

### PR DESCRIPTION
**Pull request by someniatko**
Split `useResultCache($boolFlag, ...)` into `enableResultCache()` and `disableResutCache()`.

To create tests for `enableResultCache()` and `disableResultCache()`, `useResultCache()`'s tests were copy-pasted, but I think that's fine.

### CODE REVIEW

File: lib/Doctrine/ORM/AbstractQuery.php [hunk 1]
```diff
@@ -583,21 +583,45 @@ public function getResultCacheDriver()
...
+    /**
+     * Enables caching of the results of this query, for given or default amount of seconds
+     * and optionally specifies which ID to use for the cache entry.
+     *
+     * @param integer|null $lifetime      How long the cache entry is valid, in seconds.
+     * @param string|null  $resultCacheId ID to use for the cache entry.
+     *
+     * @return static This query instance.
+     */
+    public function enableResultCache($lifetime = null, $resultCacheId = null)
```
Comment (greg0ire) [inline lib/Doctrine/ORM/AbstractQuery.php]:
```suggestion
    public function enableResultCache(?int $lifetime = null, ?string $resultCacheId = null): self
```

File: lib/Doctrine/ORM/AbstractQuery.php [hunk 2]
```diff
@@ -583,21 +583,45 @@ public function getResultCacheDriver()
...
 
+        return $this;
+    }
+
+    /**
+     * Disables caching of the results of this query.
+     *
+     * @return static This query instance.
+     */
+    public function disableResultCache()
```
Comment (greg0ire) [inline lib/Doctrine/ORM/AbstractQuery.php]:
```suggestion
    public function disableResultCache(): sef
```

**Review by greg0ire — CHANGES_REQUESTED** [review]

**Comment by greg0ire** [comment]
You will probably need to rebase as your branch contains commits that you did not author.

**Comment by someniatko** [comment]
okay, will work on it later today

**Comment by someniatko** [comment]
created new MR: https://github.com/doctrine/orm/pull/7701

**Pull request closed without merging**